packages/server/dataloader/customLoaderMakers.ts

```diff
@@ -127,15 +128,18 @@ export const userTasks = (parent: RethinkDataLoader) => {
       const r = await getRethink()
       const userIds = keys.map(({userId}) => userId)
       const teamIds = Array.from(new Set(keys.flatMap(({teamIds}) => teamIds)))
+      const archived = keys.map(({archived}) => archived).reduce((_, currVal) => currVal)
```

Comment:
this isn't gonna work.
if User 1 requests all their archived tasks & User 2 requests all their unarchived tasks, this will just return unarchived tasks for both (or maybe not, since the order is non-deterministic). Archived items also need to be sorted in order from newest to oldest.

Also, users like me have thousands of archived tasks, which means this query is going to be very expensive.

Here's what I suggest:
- add `first, after` to the `UserTasksKey`, this will be passed in via the resolver
- deep equal & dedupe key objects (ie remove keys that share the exact same values) `{userId, teamIds, archived, first, after}`
- for each deduped key, make a paginated query to the DB using `between`. see https://github.com/ParabolInc/parabol/blob/3935ArchivedCheckbox/packages/server/graphql/queries/archivedTasksCount.ts#L29 
- order the query by `updatedAt`
- use `limit(first + 1)` on the query so the DB doesn't have to return 1000s. if `first == 20` and it returns 21, then you know `hasNextPage === true`